### PR TITLE
add a header line to loudly indicate sudo editing

### PR DESCRIPTION
**Pull request by jpkotta**

### CODE REVIEW

**Comment by marsam** [comment]
Thanks, but I don't think the function `sudo-edit-set-header` belong to this package.  We could add it to the readme as a recommendation though.

**Comment by nflath** [comment]
Why not? Seems like it would be fine to me.

**Comment by jpkotta** [comment]
How about turning it into a global minor mode?

**Comment by marsam** [comment]
> Why not? Seems like it would be fine to me

`sudo-edit` is just a toggle for tramp, modifying `find-file-hook` o `dired-mode-hook` without asking the user might be invasive.

> How about turning it into a global minor mode?

I'm in favor of this

File: sudo-edit.el [hunk 1]
```diff
@@ -99,6 +99,36 @@ attention to case differences."
...
+This function is suitable to add to `find-file-hook' and `dired-file-hook'."
+  (when (string-equal
+         (file-remote-p (or buffer-file-name default-directory) 'user)
+         "root")
+    (setq header-line-format
+          (propertize "--- WARNING: EDITING FILE AS ROOT! %-"
+                      'face 'sudo-edit-header-face))))
+
+;;;###autoload
+(define-minor-mode sudo-indicator-mode
```
Comment (marsam) [inline sudo-edit.el]: Would you mind adding the following change, before I merge it
```diff
diff --git a/sudo-edit.el b/sudo-edit.el
index 8f9abd7..92ad5bb 100644
--- a/sudo-edit.el
+++ b/sudo-edit.el
@@ -116,13 +116,14 @@ This function is suitable to add to `find-file-hook' and `dired-file-hook'."
                       'face 'sudo-edit-header-face))))
 
 ;;;###autoload
-(define-minor-mode sudo-indicator-mode
+(define-minor-mode sudo-edit-indicator-mode
   "Indicates editing as root by displaying a message in the header line."
   :global t
   :lighter nil
   :group 'sudo-edit
+  :require 'sudo-edit
 
-  (if sudo-indicator-mode
+  (if sudo-edit-indicator-mode
       (progn
         (add-hook 'find-file-hook #'sudo-edit-set-header)
         (add-hook 'dired-mode-hook #'sudo-edit-set-header))
```

**Comment by marsam** [comment]
@jpkotta thanks for humor me

**Comment by jpkotta** [comment]
@marsam no problem, your requests were completely reasonable.

**Comment by nflath** [comment]
Thanks a lot!

**Pull request merged**